Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift

```diff
@@ -388,6 +406,30 @@ open class _ChatMessageListVC<ExtraData: ExtraDataTypes>:
         updateNavigationTitle()
     }
 
+    open func channelController(
+        _ channelController: _ChatChannelController<ExtraData>,
+        didChangeTypingMembers typingMembers: Set<_ChatChannelMember<ExtraData.User>>
+    ) {
+        let typingMembersWithoutCurrentUser = typingMembers.filter { $0.id != self.channelController.client.currentUserId }
+        showTypingIndicatorIfNeeded(typingMembers: typingMembersWithoutCurrentUser)
+    }
+
+    private func showTypingIndicatorIfNeeded(typingMembers: Set<_ChatChannelMember<ExtraData.User>>) {
+        if typingMembers.isEmpty {
+            typingIndicatorView.isHidden = true
+        } else {
+            // If we somehow cannot fetch any member name, we simply show that `Someone is typing`
+            guard let member = typingMembers.first(where: { user in user.name != nil }), let name = member.name else {
+                typingIndicatorView.content = L10n.MessageList.TypingIndicator.typingUnknown
+                typingIndicatorView.isHidden = false
+                return
+            }
+            
+            typingIndicatorView.content = L10n.MessageList.TypingIndicator.users(name, typingMembers.count - 1)
+            typingIndicatorView.isHidden = false
+        }
+    }
```

Comment:
IMO I think we should avoid as much as possible using "ifNeeded" methods. Although those are used on native iOS APIs, they are usually related to layout. I would say this is a bad practice for business logic because it hides the condition of when the method is executed, it is less flexible and customizable in our case. If we just do: 

```Swift
        ....
        if typingMembers.isEmpty {
            typingIndicatorView.isHidden = true
        } else {
            showTypingIndicator(typingMembers: typingMembersWithoutCurrentUser)
        }
        ....

    open func showTypingIndicator(typingMembers: Set<_ChatChannelMember<ExtraData.User>>) {
            // If we somehow cannot fetch any member name, we simply show that `Someone is typing`
            guard let member = typingMembers.first(where: { user in user.name != nil }), let name = member.name else {
                typingIndicatorView.content = L10n.MessageList.TypingIndicator.typingUnknown
                typingIndicatorView.isHidden = false
                return
            }
            
            typingIndicatorView.content = L10n.MessageList.TypingIndicator.users(name, typingMembers.count - 1)
            typingIndicatorView.isHidden = false
    }
```

With this, we can also make the `showTypingIndicator` function `open` and the customer can customize the indicator style.

Reply:
And also exporting `typingIndicatorView.isHidden = true` to 
```Swift
open func hideTypingIndicator() {
   typingIndicatorView.isHidden = true
}
```
In case the client wants to customize the hiding

Reply:
Wow thanks, this is actually great solution and makes it much more customizable... Thank you!